internal/clients/http/gcp/gcp_client.go

```diff
@@ -92,6 +92,55 @@ func (c *gcpClient) newInstancesClient(ctx context.Context) (*compute.InstancesC
 	return client, nil
 }
 
+func (c *gcpClient) NewInstanceTemplates(ctx context.Context) (*compute.InstanceTemplatesClient, error) {
+	client, err := compute.NewInstanceTemplatesRESTClient(ctx, c.options...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create GCP regions client: %w", err)
+	}
+	return client, nil
+}
+
+func (c *gcpClient) ListLaunchTemplates(ctx context.Context) ([]*clients.LaunchTemplate, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "ListLaunchTemplates")
+	defer span.End()
+
+	logger := logger(ctx)
+	logger.Trace().Msgf("Listing launch templates")
+
+	templatesClient, err := c.NewInstanceTemplates(ctx)
+	if err != nil {
+		logger.Error().Err(err).Msg("Could not get instances client")
+		return nil, fmt.Errorf("unable to get instances client: %w", err)
+	}
+	defer templatesClient.Close()
+
+	req := &computepb.AggregatedListInstanceTemplatesRequest{
+		Project: c.auth.Payload,
+	}
+	templatesList := make([]*clients.LaunchTemplate, 0)
```

Comment:
This is same as

    var templatesList []*clients.LaunchTemplate

I think. Function `append` will automatically allocate new slice when value is `nil`. This is generally preferred in Go codebases I have seen so far.